### ISSUE TITLE
No Pod Metrics

### ISSUE THREAD

**Issue by bobymicroby** [kubernetes/heapster#1154]
Hello I am running kubernetes 1.2 bare metal.

I have deployed heapster & grafana in various ways. At first i have tried using the yaml files in 
https://github.com/kubernetes/heapster/tree/master/deploy/kube-config/influxdb 
It was working for cluster metrics, but no pod metrics.

Then I have build the heapster image from master, i have included the  grafana dashboards in the grafana image allowing http://monitoring-influxdb:8086?**withfields=true** and I have deployed heapster again with 

``` yaml

apiVersion: v1
kind: ReplicationController
metadata:
  labels:
    k8s-app: heapster
    name: heapster
    version: v6
  name: heapster
  namespace: kube-system
spec:
  replicas: 1
  selector:
    k8s-app: heapster
    version: v6
  template:
    metadata:
      labels:
        k8s-app: heapster
        version: v6
    spec:
      containers:
      - name: heapster
        image: localhost:5000/novarto/heapster:v1.1.0-beta1
        imagePullPolicy: Always
        command:
        - /heapster
        - --source=kubernetes:http://10.10.0.137:8080?inClusterConfig=false
        - --sink=influxdb:http://monitoring-influxdb:8086?withfields=true

```

Cluster dashboard was working, Pods do not.

This is the log output from the heapster pod

```
I0511 15:03:05.014459       1 manager.go:98] Querying source: kubelet_summary:10.10.0.137:10255
I0511 15:03:05.015422       1 manager.go:98] Querying source: kubelet_summary:10.10.0.12:10255
I0511 15:03:05.030903       1 manager.go:152] ScrapeMetrics: time: 30.588287ms size: 6
I0511 15:03:05.030937       1 manager.go:154]    scrape  bucket 0: 2
I0511 15:03:05.030945       1 manager.go:154]    scrape  bucket 1: 0
I0511 15:03:05.030968       1 manager.go:154]    scrape  bucket 2: 0
I0511 15:03:05.030976       1 manager.go:154]    scrape  bucket 3: 0
I0511 15:03:05.030999       1 manager.go:154]    scrape  bucket 4: 0
I0511 15:03:05.031018       1 manager.go:154]    scrape  bucket 5: 0
I0511 15:03:05.031036       1 manager.go:154]    scrape  bucket 6: 0
I0511 15:03:05.031052       1 manager.go:154]    scrape  bucket 7: 0
I0511 15:03:05.031093       1 manager.go:154]    scrape  bucket 8: 0
I0511 15:03:05.031117       1 manager.go:154]    scrape  bucket 9: 0
```

And this is the output of http://10.10.0.12:10255

![image](https://cloud.githubusercontent.com/assets/927575/15186109/6a221ad8-17a4-11e6-8925-91d072ad5b9f.png)

Do i have to enable something on the Pods so I can have metrics from them ?

**Comment by mwielgus** [comment]
It is quite strange but it seems that heapster gets too few data. Can you curl heapster_ip:8082/api/v1/model/debug/allkeys  as described here https://github.com/kubernetes/heapster/blob/master/docs/debugging.md ?

**Comment by devlina** [comment]
@mwielgus I am facing similar issues, can not see any pod metrics in grafana [#26769].(https://github.com/kubernetes/kubernetes/issues/26769) Looks like I do not have api/v1/model/debug/allkeys endpoint defined. Getting logs at /metrics endpoint though.
A side thought: I am running cluster on ubuntu, do you think platform might have something to do with the issue. Would moving to CoreOS help?

**Comment by somejfn** [comment]
I think I have the same issue but do not have a working heapster setup to compare with.... what would be a typical output of <heapster scv>/api/v1/model/debug/allkeys  ?

Here's mine:
[
  "node:10.3.22.70/container:docker-daemon",
  "node:10.3.22.72/container:kubelet",
  "node:10.3.22.72/container:docker-daemon",
  "node:10.3.22.73",
  "node:10.3.22.73/container:docker-daemon",
  "node:10.3.22.73/container:kubelet",
  "node:10.3.22.70",
  "node:10.3.22.70/container:kubelet",
  "node:10.3.22.72",
  "cluster"
 ]


On k8s v1.2.4, coreos alpha 1097 and docker 1.11

**Comment by linhbombk** [comment]
Any updates? I'm facing the similar issue with heapster can collect metrics nodes level but namespace,pod not.
The output of heapster_ip:8082/api/v1/model/debug/allkeys only shows cluster and node infomation.
I can query namespace,pod,container... directly from kubernetes master.
k8s v1.2,heapter image: canary,docker 1.11.2,Cenos 7.

**Comment by suyogbarve** [comment]
Any update I am facing same problem under similar configuration

**Comment by nwind21** [comment]
I have a similar issue.  I used kube-aws to setup which by default has a heapster monitoring, tried the kubernetes/heapster instructions but that didn't pick up the pods nor cluster, and ended up taking the yaml's from kubernetes/cluster/addons/cluster-monitoring/influxdb.

I got node cluster data but _no namespaces nor pods_ which seems to match what others are seeing on this issue report.  I checked cAdvisor and I think everything looks good.

- k8s: 1.3.6
- coreos: 1122.3.0
- docker: 1.10.3
- heapster: gcr.io/google_containers/heapster:v1.1.0
- heapster source:  kubernetes.summary_api:''

Note, I did try heapster:canary and it didn't seem to scrape data to influx.  I also tried different sources, but my understanding of the sources isn't clear so it could've been something else.

**Comment by autostatic** [comment]
Having the same issue here with Kubernetes 1.2.4 while the same deployment works fine with Kubernetes 1.4.0 and higher.

**Comment by piosz** [comment]
There was a lot of problems with gathering metrics on CoreOS (see https://github.com/kubernetes/kubernetes/issues/30939, https://github.com/kubernetes/kubernetes/issues/27194). Most of them should be fixed in Kubernetes 1.4. Please try to upgrade. I'm closing now, please reopen in case of more issues.

**Comment by ajaybhande** [comment]
I see same issue with Kubernetes 1.4.6 and Heapster 1.2.0. The /api/v1/model/debug/allkeys shows only node and cluster.
[
  "node:10.22.152.159",
  "cluster"
 ]

Heapster logs show something like below for Pod and proper number for Node:
reflector.go:403] k8s.io/heapster/metrics/heapster.go:319: Watch close - *api.Pod total 0 items received

What could be problem?
Thank you!
